WoopraSDK/WTracker.swift

```diff
@@ -12,7 +12,7 @@ public class WTracker: WPropertiesContainer {
     // MARK: - Public properties
     // Identifies which project environment your sending this tracking request to. E.g. http://yourproject.com
     @objc dynamic public var domain: String?
-    @objc dynamic public var visitor: WVisitor!
```

Comment:
TBD: To avoid runtime crashes, we can either
1. provide a default visitor. It could be anonymousVisitor
2. make this property as optional
3. ask for a visitor while initializing it

Reply:
just wondering, as it has the `instance.visitor = WVisitor.anonymousVisitor()` at line 50 in the `shared` function. Is it possible for the visitor to be nil?

Reply:
Ah, I suppose if we don't make this change, it will fail the test case.

Reply:
Line 50 only assigns the visitor to the singleton `shared` only. The visitor is not assigned for any WTracker initialized directly. Another way is that we make the init method private so that the users can not init a WTracker on their own. They can only use the `shared` instance.

Reply:
I suppose using a singleton is the only way we can expect the user to proceed, but to make it testable, maybe we will need to use dependency injection.

Reply:
I think we can make the init() internal access therefore we can test meanwhile prohibiting a user from using it.